pkgs/tools/package-management/nix/default.nix

```diff
@@ -184,6 +184,12 @@ in lib.makeExtensible (self: ({
     self_attribute_name = "nix_2_23";
   };
 
+  nix_2_24 = common {
+    version = "2.24.1";
+    hash = "sha256-3yFEvUDPB7GlCMI9I5VV+HXMVOT38h3lnw01nIXU2F4=";
+    self_attribute_name = "nix_2_24";
+  };
```

Comment:
You need the SDK stuff from `nixVersions.git` for this to work on `x86_64-darwin`.

Reply:
Thanks. Added.